tests/functional/admin_controller_test.php

```diff
@@ -125,4 +125,42 @@ public function test_notifications()
 		$crawler = self::request('GET', "index.php?&sid={$this->sid}");
 		$this->assertContainsLang('BOARDRULES_NOTIFICATION', $crawler->filter('.notification-title')->text(), 'The notification was not found in the notifications list');
 	}
+
+	/**
+	 * Test Board Rules ACP Settings
+	 */
+	public function test_acp_settings_and_logs()
+	{
+		$this->login();
+		$this->admin_login();
+
+		$this->add_lang_ext('phpbb/boardrules', 'info_acp_boardrules');
+		$crawler = self::request('GET', "adm/index.php?i=-phpbb-boardrules-acp-boardrules_module&mode=settings&sid={$this->sid}");
+		$form = $crawler->selectButton('submit')->form();
+		$crawler = self::submit($form);
+		$this->assertContainsLang('ACP_BOARDRULES_SETTINGS_CHANGED', $crawler->text());
+
+		// Confirm the log entry has been added correctly
+		$crawler = self::request('GET', 'adm/index.php?i=acp_logs&mode=admin&sid=' . $this->sid);
+		$this->assertContains(strip_tags(html_entity_decode($this->lang('ACP_BOARDRULES_SETTINGS_LOG'), ENT_COMPAT, 'UTF-8')), $crawler->text());
```

Comment:
`$this->assertContainsLang('ACP_BOARDRULES_SETTINGS_LOG', $crawler->text());` should be enough